### PR TITLE
Feature require auth

### PR DESCRIPTION
**Pull request by whwalter**
This requires github signing for github webhooks and bearer tokens for webhooks that have the ingress annotations.

Unit tests are provider for the validate function. 

Testing on kind can be performed with the following steps:

1. start with a working kind cluster with istio as the primary ingress controller and argo-events running
2. run the make local target from the root of the repo
3. apply the examples/testers/gw.yaml, examples/testers/event-bus.yaml, and examples/testers/demo-secret.yaml  manifest files
4. after the event bus pods are ready apply and of the examples/testers/es-{github,jira,officeips}.yaml manifests.
5. confirm the following resources are created for each manifest

* VirtualService routing-rules/devops-argo-tester-{github.jira,officeips}
* AuthorizationPolicy routing/devops-argo-tester-{github.jira,officeips}

7. A webhook can be passed to each event source via curl. The below example is for the jira tester.
```
curl -k -XPOST  -H 'Authorization: Bearer thisisanexample --data '{ "msg": "hello-world" }' https://webhooks.example.com/devops/argo-tester-jira/example
```
8. Verify that the event is received on the event bus via pod logs for the event source.

### CODE REVIEW

File: internal/admission/eventsource/handler_test.go [hunk 1]
```diff
@@ -155,3 +168,131 @@ func TestEventSourceHandler(t *testing.T) {
 
 	}
 }
+
+func TestValidateGithubEventSource(t *testing.T) {
```
Comment (yuzhouliu9) [inline internal/admission/eventsource/handler_test.go]:
```suggestion
func TestValidateEventSource(t *testing.T) {
```

File: examples/testers/es-github.yaml [hunk 2]
```diff
@@ -11,8 +11,17 @@ spec:
...
+        method: POST
+        port: "8080"
+      repositories:
+        - owner: kanopy-platform
+          names:
+            - argoslower
+      webhookSecret:
+        name: argoslower-examples
+        key: token
+        optional: true
```
Comment (yuzhouliu9) [inline examples/testers/es-github.yaml:27]: If we're enforcing that a `webhookSecret` or `authSecret` must be defined, should `optional: false` also be enforced?

However in testing, it seems like `optional` doesn't take effect, and it always checks for authorization header.

Comment (whwalter) [reply examples/testers/es-github.yaml:27]: The pod wil crashloop if the secret isn't there. This was just there to make the convergence for testing simpler.

Comment (whwalter) [reply examples/testers/es-github.yaml:27]: This is a https://pkg.go.dev/k8s.io/api/core/v1#SecretKeySelector so the 'optional' is the secret existing for pod scheduling not optionally requiring auth.

Comment (yuzhouliu9) [reply examples/testers/es-github.yaml:27]: Oh I see

**Review by yuzhouliu9 — APPROVED** [review]

**Pull request merged**